### PR TITLE
Set User Agent for FFmpeg calls

### LINKED ISSUE #4369: Set `-user_agent` in `ffmpeg` calls
**Describe what you are trying to accomplish and why in non technical terms**
When spoofing who is connecting to my cameras, I can easily distinguish some like Home Assistant, or Google Chrome. But Frigate is also connected to them, but since `-user_agent` is not set, it's not easi to distinguish.

image

**Describe the solution you'd like**
This can be set with the `-user_agent` ffmpeg flag. In order of usefulness but also in order of difficulty to implement:

1. `-user_agent 'Frigate'` already much better than now
2. `-user_agent 'Frigate/0.11.1'`
3. `-user_agent 'Frigate/0.11.1 FFmpeg'`
4. `-user_agent 'Frigate/0.11.1 FFmpeg/3.1.2'`

I guess that, with the FFmpeg version in the string we don't need the `Lavf59.27.100` either. But it wouldn't hurt to even add the `Lavf59.27.100`, I guess.

**Describe alternatives you've considered**
I can probably set this by myself using the args, but I think this deserves to come by default.

**Additional context**
- Refs https://github.com/AlexxIT/go2rtc/issues/106

### PR DESCRIPTION
**Pull request by felipecrs**
Closes #4369

### CODE REVIEW

**Comment by felipecrs** [comment]
Another option I thought of was to set them here, disallowing users to override:

image

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (NickM-27) [inline .vscode/launch.json:1]: Need to remove this too along with the ignore

**Review by NickM-27 — CHANGES_REQUESTED** [review]

**Comment by NickM-27** [comment]
> Another option I thought of was to set them here, disallowing users to override:
> 
> image
> 

I see no reason to make it mandatory, users should be able to change it if they want to

File: .vscode/launch.json (hunk 1 above)
Comment (felipecrs) [reply .vscode/launch.json:1]: This file was meant to be tracked bt git. See:

https://github.com/github/gitignore/blob/main/Global/VisualStudioCode.gitignore

It allows anyone who is using the devcontainer to easily debug it by simply pressing F5.

**Comment by felipecrs** [comment]
> I see no reason to make it mandatory, users should be able to change it if they want to

Cool, I agree.

**Review by NickM-27 — APPROVED** [review]

File: docs/docs/configuration/index.md [hunk 2]
```diff
@@ -142,7 +142,7 @@ birdseye:
 # Optional: ffmpeg configuration
 ffmpeg:
   # Optional: global ffmpeg args (default: shown below)
-  global_args: -hide_banner -loglevel warning
+  global_args: -hide_banner -loglevel warning -user_agent "FFmpeg Frigate"
```
[the remark this answers is not in the document]
Comment (NickM-27) [reply docs/docs/configuration/index.md:145]: I don't think so, the amount of users running multiple frigate instances where the version number would be needed will be quite few IMO

[the remark this answers is not in the document]
Comment (felipecrs) [reply docs/docs/configuration/index.md:145]: I concur. :)

File: .gitignore [hunk 3]
```diff
@@ -2,7 +2,8 @@
 *.pyc
 *.swp
 debug
-.vscode
+.vscode/**
```
Comment (felipecrs) [inline .gitignore:5]:
```suggestion
.vscode/*
```

File: frigate/util.py [hunk 4]
```diff
@@ -888,7 +889,7 @@ def vainfo_hwaccel() -> sp.CompletedProcess:
 
 def get_ffmpeg_arg_list(arg: Any) -> list:
     """Use arg if list or convert to list format."""
-    return arg if isinstance(arg, list) else arg.split(" ")
+    return arg if isinstance(arg, list) else shlex.split(arg)
```
Comment (felipecrs) [inline frigate/util.py:892]: @NickM-27 you may want to re-review this PR after this change.

Comment (NickM-27) [reply frigate/util.py:892]: yeah I saw that, can you add a test to the test_ffmpeg_presets.py which manually sets the inputs args as default args but in a single string?

Comment (felipecrs) [reply frigate/util.py:892]: Done, I'm even testing the string with an `"arg with space"` in it as well.

**Review by NickM-27 — APPROVED** [review]

**Pull request merged**